LFS-938: Move the BioPortalApiKey serverside code into a separate API component

veronikaslc opened the pull request:
Also fixes:
[LFS-990: ICD* vocabularies can not longer be installed either locally or via Bioportal](https://phenotips.atlassian.net/browse/LFS-990)
[LFS-396: Failed to Install Cancer Chemoprevention Ontology](https://phenotips.atlassian.net/browse/LFS-396)
[LFS-391: Failed to Install International Classification of Functioning, Disability and Health](https://phenotips.atlassian.net/browse/LFS-391)
[LFS-389: Failed to install Plant Trait Ontology](https://phenotips.atlassian.net/browse/LFS-389)

Commit:
LFS-938: Move the BioPortalApiKey serverside code into a separate API component

sdumitriu commented on modules/vocabularies/src/main/java/ca/sickkids/ccm/lfs/vocabularies/internal/DefaultBioPortalApiKeyManager.java:
```diff
@@ -0,0 +1,89 @@
+    {
+        String apiKey = System.getenv(APIKEY_ENVIRONMENT_VARIABLE);
+        apiKey = StringUtils.isBlank(apiKey) ? "" : apiKey;
+        LOGGER.info("BioPortal API key as set in the OS environment: [{}]", apiKey);
```
Change to `LOGGER.debug` instead.

sdumitriu commented on modules/vocabularies/src/main/java/ca/sickkids/ccm/lfs/vocabularies/internal/DefaultBioPortalApiKeyManager.java:
```diff
@@ -0,0 +1,89 @@
+            Resource res = this.rrf.getThreadResourceResolver().getResource(resourcePath);
+            Node keyNode = res.adaptTo(Node.class);
+            apiKey = keyNode.getProperty("key").getString();
+            LOGGER.info("BioPortal API key as set in the BioportalApiKey node: [{}]", apiKey);
```
Change to `LOGGER.debug` instead.

sdumitriu commented on modules/vocabularies/src/main/java/ca/sickkids/ccm/lfs/vocabularies/BioPortalApiKeyManager.java:
```diff
@@ -0,0 +1,44 @@
+ */
+public interface BioPortalApiKeyManager
+{
+    String getAPIKey();
```
Add javadoc: `Retrieves the BioPortal API key configured in the system. @return the configured key, or the empty string if no key is configured.`

sdumitriu commented on modules/vocabularies/src/main/java/ca/sickkids/ccm/lfs/vocabularies/BioPortalApiKeyManager.java:
```diff
@@ -0,0 +1,44 @@
+package ca.sickkids.ccm.lfs.vocabularies;
+
+/**
+ * Interface for BioPortal API key component.
```
```suggestion
 * Service for BioPortal API key management.
```

Commit:
LFS-938: Move the BioPortalApiKey serverside code into a separate API component
Addressed codereview comments

sdumitriu committed:
[cleanup] Applied codestyle

sashaandjic commented:
cannot build this one 
![image](https://user-images.githubusercontent.com/23243244/111211107-63c64100-85a4-11eb-8e75-4ece5ea7aaad.png)

marta- commented:
> cannot build this one

@sdumitriu 's latest commit with code style improvements should have fixed the checkstyle failures.

sashaandjic commented:
Attempting to installing any  ICD*  vocabulary brings login dialog infinitive number of times

sashaandjic commented:
I had no issue installing HACNCESTRO, HP and ORDO from BioPortal; and HPO from local files.
However, uninstalling e.g. HANCESTRO ends up with the white screen of death
![image](https://user-images.githubusercontent.com/23243244/111223867-1651d000-85b4-11eb-95df-bc073c0b5fce.png)

marta- commented:
> However, attempting to uninstall a vocabulary ( e.g. HANCESTRO) brings the white screen of death.

That's "expected", LFS-1020, out of scope for this task.

sdumitriu committed:
LFS-990: ICD* vocabularies can not longer be installed either locally or via Bioportal

sdumitriu committed:
LFS-938: Move the BioPortalApiKey serverside code into a separate API component
Fixed error logged when they key is configured in the environment, not in the repo

sashaandjic commented:
Right after installing HPO form a local file I get all these orange buttons. After I refreshed the page, there were all gone.
![image](https://user-images.githubusercontent.com/23243244/111233444-5f5d5080-85c3-11eb-8229-9af6f6e66e36.png)

Commit:
[misc] Better error handling for vocabularies install and uninstall
